voyager-navigator/src/commonMain/kotlin/cafe/adriel/voyager/navigator/lifecycle/NavigatorLifecycleStore.kt

```diff
@@ -39,5 +41,6 @@ public object NavigatorLifecycleStore {
 
     public fun remove(navigator: Navigator) {
         owners.remove(navigator.key)?.forEach { it.value.onDispose(navigator) }
+        ScreenLifecycleStore.remove(screen = NavigatorScreen(key = navigator.key))
```

Comment:
NavigatorLifecycleStore should not have this responsibility.

Reply:
I need a unique place where I know that Navigator is been disposed to clean that resource [here](https://github.com/adrielcafe/voyager/blob/ef893538390d4a94285ec2a88e3cb831aa4bb895/voyager-navigator/src/androidMain/kotlin/cafe/adriel/voyager/navigator/NavigatorViewModel.kt#L37)

Reply:
Can you do like ScreenModel API does?
https://github.com/adrielcafe/voyager/blob/main/voyager-screenmodel/src/commonMain/kotlin/cafe/adriel/voyager/core/model/NavigatorScreenModel.kt#L18

Lazy inserting this behavior only when the API is first used.